core/src/client/mod.rs

```diff
@@ -283,6 +286,41 @@ pub trait DatabaseClient {
     fn database(&self) -> Arc<KeyValueDB>;
 }
 
+pub trait StateClient: BlockInfo {
```

Comment:
Why do you extract StateClient trait?

Reply:
```tendermint/mod.rs``` holds trait object of Client. So these methods should be in some trait.